GX-TimeFrequency/test/test_gx_minimax_grid.py

```diff
@@ -0,0 +1,116 @@
+""" Inputs and assertions for test_gx_minimax_grid.f90
+
+Would be better if one had python API to GX, and avoid the need for this intermediate binary.
+Indeed, MUCH of the complication regarding running tests comes from this.
+
+Issues and solutions:
+---------------------
+Problem
+* python needs to know where the test binary is located.
+Solution
+* The robust but less-than-ideal approach solution is to have python recursively
+  search through the directories for it.
+  This work very well if one defines the tests to be run from a given build
+  directory.
+
+Problem
+* Where should the python test write the input data for the fortran binary?
+Solution
+* This is solved using tmp_path, which gives an absolute path.
+
+
+* Path for passing to the fortran binary     () This is solved using tmp_path, which gives an absolute path
+* where to read the result from              () This is solved using tmp_path, which gives an absolute path
+"""
+import numpy as np
+from pathlib import Path
+
+from pygreenx.run import BinaryRunner, BuildType
+from pygreenx.utilities import find_test_binary
+
+binary_name = "test_gx_minimax_grid.exe"
+
+
+def mock_file(tmp_path, inputs_str):
+    """ Write an inputs string to file.
+    :param tmp_path:
+    :param inputs_str:
+    :return:
+    """
+    # Write test inputs to file
+    file: Path = tmp_path / "inputs.dat"
```

Comment:
This does not look live valid Python syntax

Reply:
It looks to me like type hinting syntax somehow, but that is only valid for function arguments

Reply:
You are correct, it is type-hinting. And you can most definitely use it for any data declaration ;) 

See its use in [dataclasses](https://docs.python.org/3/library/dataclasses.html), for example.

Reply:
Oh, that is really nice!